packages/suite/src/components/suite/layouts/SuiteLayout/Sidebar/QuickActions/Update/useUpdateStatus.ts

```diff
@@ -70,15 +73,23 @@ export const useUpdateStatus = (): UpdateStatusData => {
     const desktopUpdate = useSelector(state => state.desktopUpdate);
 
     const isDeviceDisconnected = device?.connected !== true;
-    const isFirmwareOutdated = device?.firmware === 'outdated';
-
     // If firmware is outdated and suite update download/check is in progress,
     // we suppress the Firmware notification as it can be there just for a second and then
     // it will be replaced with Suite update notification
     const isSuiteUpdateInProgress = [UpdateState.Downloading, UpdateState.Checking].includes(
         desktopUpdate.state,
     );
 
+    const { releaseConditions: { environment, shouldBeOffered } = {} } =
+        device?.firmwareReleaseConfig || {};
+
+    const isValidSuiteVersion =
+        !isDeviceDisconnected &&
+        versionUtils.isNewerOrEqual(getSuiteVersion(), environment?.min_suite_version);
```

Comment:
isValidSuiteVersion variable name based on isDeviceDisconnected  does not sounds logic. nitpick

Reply:
The reason of this is that when device is not connected `environment?.min_suite_version` is undefined and when you start the process of flashing firmware since it reboots from Firmwere mode to Bootloader mode there is a moment when device is "disconnected" and therefore that fails.

I could add a comment there, or maybe find other solution.